### ISSUE TITLE
Export `Locale` types to create custom `Locale` interface 

### ISSUE THREAD

**Issue by NuroDev** [QuiiBz/next-international#7]
Currently the recommended practice is to create a locale `as const`, export it, import it as a type & use it for the `createI18n` generic.

In future to change all other types you need to change that objects structure to change the type which doesn't seem right.

My proposal to make this better is to add the `Locale` & `LocaleValue` types in [`types.ts`](https://github.com/QuiiBz/next-international/blob/main/packages/next-international/src/types.ts#L1-L2) to the root export such that you can create a general purpose interface or type to manage your locales.

## Example

```typescript
// locales/index.ts
import { createI18n } from 'next-international';

import type { Locale as BaseLocale } from 'next-international';

interface MyCustomLocale extends BaseLocale {
  hello: string;
}

export const {
  defineLocale,
  useI18n,
  // ...
} = createI18n<MyCustomLocale>({
  en: () => import('./en'),
});
```

```typescript
// locales/en.ts
import { defineLocale } from '.';

export default defineLocale({
  hello: 'Hello',
})
```

**Comment by QuiiBz** [comment]
I like this idea. It can be taken even further by overriding an interface declared in `next-international`, to avoid the generic on `createI18n`, and import `defineLocale` directly from the package:

```ts
// locales/index.ts
import { createI18n } from 'next-international';

declare module 'next-international' {
  interface Locale {
    hello: string
  }
}

export const {
  useI18n,
  // ...
} = createI18n({
  en: () => import('./en'),
});
```

```ts
// locales/en.ts
import { defineLocale } from 'next-international';

export default defineLocale({
  hello: 'Hello',
})
```

The downside is that you must duplicate all your keys to define them in the type (`MyCustomLocale` in your example), instead of writing them only in the locale files.

But it should also work with JSON locales, so maybe we can explicitly type the parameters to get type-safety while using JSON files:

```json
// locales/en.json
{
  "hello": "Hello {param}, this is {anotherParam}"
}
```

```ts
// locales/index.ts
interface Locale {
  hello: '{param} {anotherParam}';
}

const { useI18n } = createI18n({
  en: () => import('./en.json'),
});

const { t } = useI18n()
// can have type-safety
t('hello', {
  param: '',
  anotherParam: '',
})
```

**Comment by QuiiBz** [comment]
Also, this would remove the need for some methods in `createI18n` that were used only to get the right type, resulting in better code-splitting.

**Comment by QuiiBz** [comment]
After trying a bit more moving in this direction, I'm not sure if it's a good idea:

- One of the goals of this lib is to avoid explicitly creating the types for the locales, which will not be the case if we move in this direction
- Adding a new key in locales should be as simple as adding it to the locales files - here, we would need to also add it to the interface in order to use it

Let me know what you think.

**Comment by NuroDev** [comment]
These are fair points. However I currently feel like offering the option to create an explicit locale interface would still be good.
The primary reason I am in favour of offering it as an option is this:

> Adding a new key in locales should be as simple as adding it to the locales files

For example, if a new hire or community member wants to contribute to a OS project & they have not used `next-international` before they may not understand that in order to change or update the locale schema they will need to update the "default locale", or whatever one is used for type inference, first in order for the type inference to be updated & other locales to require the new locales.

Some people may like the current method, and that is fine, but I personally find it odd to need to update the data structure of one file to change others & think it would be better to offer explicit types as an option.

Currently there is a simple way around this by just redefining the `LocaleValue` & `Locale` types:
```typescript
type LocaleValue = string | number;

type BaseLocale = Record<string, LocaleValue>;
```
While this works it would be better if these were exported by the package to be consumed in case of any changes during dependency upgrades.

**Comment by QuiiBz** [comment]
I understand better the issue. We can definitely export those two types and add an example in the README on how to set up an explicit interface for locales.

**Comment by QuiiBz** [comment]
Created #10, feel free to check the PR and leave me your feedback.

**Comment by QuiiBz** [comment]
Landed in version 0.0.5 (#10), thanks!
